format/udf-spec.md

```diff
@@ -0,0 +1,285 @@
+| *optional*  | `doc`             | `string`               | Documentation string.                                                                                           |
+
+Notes:
+1. When `secure` is `true`,
+    - Engines **SHOULD NOT** expose the function definition through any inspection (e.g., `SHOW FUNCTIONS`).
+    - Engines **SHOULD** ensure that execution does not leak sensitive information through any channels, such as error messages, logs, or query plans.
+   
+### Overload
+
+Function overloads allow multiple implementations of the same function name with different signatures. Each overload has
+the following fields:
+
+| Requirement | Field name                 | Type                                                   | Description                                                                                   |
+|-------------|----------------------------|--------------------------------------------------------|-----------------------------------------------------------------------------------------------|
+| *required*  | `overload-id`              | `long`                                                 | Monotonically increasing identifier of this function overload.                                |
+| *required*  | `parameters`               | `list<parameter>`                                      | Ordered list of [function parameters](#parameter). Invocation order **must** match this list. |
+| *required*  | `return-type`              | `string` (Iceberg data type)                           | Return type. Example: `"string"`, `"struct<...>"`.                                            |
+| *required*  | `versions`                 | `list<overload-version>`                               | [Versioned implementations](#overload-version) of this overload.                              |
+| *required*  | `current-overload-version` | `long`                                                 | Monotonically increasing identifier of the current overload version.                          |
+| *optional*  | `function-type`            | `string` (either `"udf"` or `"udtf"`, default `"udf"`) | If `"udtf"`, `return-type` must be a `struct` describing the output schema.                   |
+| *optional*  | `doc`                      | `string`                                               | Documentation string.                                                                         |
+
+### Parameter
+| Requirement | Field  | Type                         | Description              |
+|-------------|--------|------------------------------|--------------------------|
+| *required*  | `name` | `string`                     | Parameter name.          |
+| *required*  | `type` | `string` (Iceberg data type) | Parameter data type.     |
+| *optional*  | `doc`  | `string`                     | Parameter documentation. |
```

Comment:
Is there a way to specify that null values are not allowed? In Iceberg nested types, each field/element/value is required or optional.

Reply:
Added one rule for inputs to clarify it.